six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp

```diff
@@ -0,0 +1,307 @@
+    for (size_t ii = 0; ii < readData.size(); ++ii)
+    {
+        std::complex<float> val(writeData[ii].real(), writeData[ii].imag());
+        if (scale)
+        {
+            val *= scaleFactors[ii / pointsPerScale];
+        }
+
+        if (val != readData[ii])
+        {
+            std::cerr << "Value mismatch at index " << ii << std::endl;
+            return false;
+        }
+    }
+    return true;
+}
+
+template<typename T>
+bool runTest(bool scale, const std::vector<std::complex<T> >& writeData)
+{
+    TempFile tempfile;
+    const size_t numThreads = sys::OS().getNumCPUs();
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<double> scaleFactors =
+            generateScaleFactors(dims.row, scale);
+    writeCPHD(tempfile.pathname(), numThreads, dims, writeData);
+    const std::vector<std::complex<float> > readData =
+            checkData(tempfile.pathname(), numThreads, scaleFactors,
+            scale, dims);
+    return compareVectors(readData, writeData, scaleFactors, scale);
+}
+
+TEST_CASE(testUnscaledInt8)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<sys::Int8_T> > writeData =
+            generateData<sys::Int8_T>(dims.area());
+    const bool scale = false;
+    TEST_ASSERT(runTest(scale, writeData))
+}
+
+TEST_CASE(testScaledInt8)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<sys::Int8_T> > writeData =
+            generateData<sys::Int8_T>(dims.area());
+    const bool scale = true;
+    TEST_ASSERT(runTest(scale, writeData));
+}
+TEST_CASE(testUnscaledInt16)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<sys::Int16_T> > writeData =
+            generateData<sys::Int16_T>(dims.area());
+    const bool scale = false;
+    TEST_ASSERT(runTest(scale, writeData))
+}
+
+TEST_CASE(testScaledInt16)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<sys::Int16_T> > writeData =
+            generateData<sys::Int16_T>(dims.area());
+    const bool scale = true;
+    TEST_ASSERT(runTest(scale, writeData));
+}
+
+TEST_CASE(testUnscaledFloat)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<float> > writeData =
+            generateData<float>(dims.area());
+    const bool scale = false;
+    TEST_ASSERT(runTest(scale, writeData))
+}
+
+TEST_CASE(testScaledFloat)
+{
+    const types::RowCol<size_t> dims(128, 128);
+    const std::vector<std::complex<float> > writeData =
+            generateData<float>(dims.area());
+    const bool scale = true;
+    TEST_ASSERT(runTest(scale, writeData));
+}
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        TEST_CHECK(testUnscaledInt8);
+        TEST_CHECK(testScaledInt8);
+        TEST_CHECK(testUnscaledInt16);
+        TEST_CHECK(testScaledInt16);
+        TEST_CHECK(testUnscaledFloat);
+        TEST_CHECK(testScaledFloat);
+        TempFile tempfile;
```

Comment:
Can you update the Python test script to call this application?  Or is something else running the unittests automatically right now?